fastify.d.ts

```diff
@@ -391,7 +402,8 @@ declare namespace fastify {
     /**
      * Add a content type parser
      */
-    addContentTypeParser(contentType: string, opts: object | AsyncContentTypeParser<HttpRequest> | ContentTypeParser<HttpRequest>, parser?: AsyncContentTypeParser<HttpRequest> | ContentTypeParser<HttpRequest>): void;
+    addContentTypeParser(contentType: string, parser?: AsyncContentTypeParser<HttpRequest> | ContentTypeParser<HttpRequest>): void;
+    addContentTypeParser(contentType: string, opts: AddContentParserOptions | AsyncContentTypeParser<HttpRequest> | ContentTypeParser<HttpRequest>, parser?: AsyncContentTypeParser<HttpRequest> | ContentTypeParser<HttpRequest>): void;
```

Comment:
Since this is using method overriding, the `opts` and `parser` parameters should be typed as:
```ts
opts: AddContentParserOptions, parser: AsyncContentTypeParser<HttpRequest> | ContentTypeParser<HttpRequest>
```
(otherwise it would be valid to pass a parser function as both the `opts` and `parser` parameters).